src/tools/compiletest/src/runtest.rs

```diff
@@ -352,13 +354,24 @@ impl<'test> TestCx<'test> {
         if self.props.should_ice {
             match proc_res.status.code() {
                 Some(101) => (),
-                _ => self.fatal("expected ICE"),
+                _ => self.fatal("test no longer crashes/triggers ICE! Please annotate it and add it as test to tests/ui or wherever you see fit"),
             }
         }
 
         self.check_forbid_output(&output_to_check, &proc_res);
     }
 
+    fn run_crash_test(&self) {
+        let pm = self.pass_mode();
+        let proc_res = self.compile_test(WillExecute::No, self.should_emit_metadata(pm));
+
+        // if a test does not crash, consider it an error
+        match proc_res.status.code() {
+            Some(101) => (),
+            _ => self.fatal("expected ICE"),
+        }
+    }
+
```

Comment:
Soo it turns out that this is not very sufficient at detecting when rustc is killed by a signal for example due to stack overflow.

There is something like `std::os::unix::process::ExitStatusExt` to get more information for cases like this, but it seems to be unix-only.

I think the best approach for now is to switch around the logic to only "deny" cases where exit status is 0 (no error) or 1 (compiler error) and accept everything else as ICE...?

```rust
        if !proc_res.status.success() {
            match proc_res.status.code() {
                Some(1 | 0) => self.fatal(&format!("expected ICE")),
                _ => (),
            }
        }
```